python/paddle/fluid/dygraph/dygraph_to_static/loop_transformer.py

```diff
@@ -160,8 +164,18 @@ def _var_nodes_to_names(self, node_set, ctx_filter_set=None):
         for node in node_set:
             if ctx_filter_set is None or type(node.ctx) in ctx_filter_set:
                 if isinstance(node, gast.Name):
+                    parent_node = self.node_to_wrapper_map[node].parent.node
```

Comment:
This condition is very similar to my `_is_call_func_name_node`, and I found my bug may be line 184:

parent_node = self.ancestor_nodes[-1] -> parent_node = self.ancestor_nodes[-2] 

The self.ancestor_nodes and _is_call_func_name_node are doing same thing as your condition. So I would recommend to remove duplicate codes. I suggest two options for you:

Option 1: fix the bug I mentioned at line 184, remove the code you added to see whether the bug is fixed. In this way we don't make static analysis in this Transformer (**Program may be slightly faster**).

Option 2: keep the code you added and help me to remove self.ancestor_nodes, _is_call_func_name_node, in this way the code is **safer** because the parent_node is analyzed by static_analysis.

Both are OK to me, you can pick one. Personally I slightly prefer Option2

Reply:
OK. I also prefer Option2. I will remove  self.ancestor_nodes.